src/test/index.ts

```diff
@@ -330,8 +415,14 @@ export const test = _baseTest.extend<TestFixtures, WorkerAndFileFixtures>({
     const allPages: Page[] = [];
     context.on('page', page => allPages.push(page));
 
+    if (_reuseBrowerContext.isEnabled())
+      _reuseBrowerContext.setContext(context);
+
     await use(context);
 
+    if (_reuseBrowerContext.isEnabled())
+      return;
```

Comment:
We are not turning off video / tracing here, looks like the logic is off.

Reply:
simplified that logic.